### DISCUSSION TITLE
STM-GAP8 CPX communication Example: Terminal Command Failure

### DISCUSSION THREAD

**Discussion by LucasTheIntern** [bitcraze/discussions#994] [Q&A] [answered]
Hello!

I'm attempting to get the stm-gap8 cpx communication example up and running on my CF and AI deck. I have completed all instructions on the ["Getting started with the AI deck"](https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/) page, including successful runs of the wifi streaming example, setting up the development environment, and cloning the gap8 examples repo. 

I am now following the instructions on the stm-gap8 cpx communication example page:
https://www.bitcraze.io/documentation/repository/aideck-gap8-examples/master/other-examples/stm_gap8_cpx/

The first step is to run the command:
`$ docker run --rm -v ${PWD}:/module bitcraze/aideck tools/build/make-example examples/other/stm_gap8_cpx all`

This yields the following output in the terminal:
`
docker: Error response from daemon: failed to create shim: OCI runtime create failed: container_linux.go:380: starting container process caused: exec: "tools/build/make-example": stat tools/build/make-example: no such file or directory: unknown.
bitcraze@ubuntu:~$ 
`

I don't know what this means... I tried variations of the command (mostly stuff with the directory), all of which gave me errors different from above. For example, I tried the following based on what seems to be the correct directory on my version of the VM:
`docker run --rm -v ${PWD}:/module bitcraze/aideck-gap8-examples/tools/build/make-example examples/other/stm_gap8_cpx all`
because I was able to find the "stm_gap8_cpx.c" file and a Makefile in that directory. 
This gave me the error:
`
Unable to find image 'bitcraze/aideck-gap8-examples/tools/build/make-example:latest' locally
docker: Error response from daemon: pull access denied for bitcraze/aideck-gap8-examples/tools/build/make-example, repository does not exist or may require 'docker login': denied: requested access to the resource is denied.
See 'docker run --help'.
`

At this point, I'm fairly certain that I'm just missing something obvious... any tips would be appreciated!

**Comment by krichardsson** [comment]
It looks like there might be a problem with docker. Does the `bitcraze/aideck` image work?

What happens if you start it with `docker run --rm -it bitcraze/aideck`? 
It should start the container and show a prompt, something similar to 
`root@2626aceb3e5a:/module#`

**Reply by LucasTheIntern** [reply]
That command seems to be working. When run in a new terminal, It returns the prompt:
`root@4c17b1c0bbeb:/module# 
`

I've been looking into docker tutorials since I'm not familiar with it... no revelations as of yet. 

I'm also uncertain that my file system is correct... some things seem to be out of place, but this is a fairly new clone of the VM that I haven't used much. I _am_  using and older version of the VM as suggested here:
https://github.com/bitcraze/bitcraze-vm/issues/66
Not sure if this matters, but it looks like that issue has been fixed... so I might try upgrading and going through the wifi streamer example again.

An example of what I mean by the oddities in the file system:

The first command on the stm-gap8 cpx example page includes the directory:
`bitcraze/aideck tools/build/make-example`
but this doesn't seem to exist on my VM. Instead, I find that file (is it the same file?) by:
`bitcraze/aideck-gap8-examples/tools/build/make-example`
since there is no folder under 'bitcraze' labeled 'aideck tools'

Thanks for your help!

**Comment by knmcguire** [comment, marked as answer]
I've tried to build  it the same way with the latest aideck-gap8-examples and it all worked for me. You should run it from the root folder of the aideck-gap8-example repo, so no cding in the example folder itself. 

```
kimberly's computer:~/Development/bitcraze/c/aideck-gap8-examples$ docker run --rm -v ${PWD}:/module bitcraze/aideck tools/build/make-example examples/other/stm_gap8_cpx all
Collecting numpy==1.22.3
  Using cached numpy-1.22.3-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (16.8 MB)
Installing collected packages: numpy
  Attempting uninstall: numpy
ERROR: imageio 2.17.0 has requirement pillow>=8.3.2, but you'll have pillow 7.2.0 which is incompatible.
    Found existing installation: numpy 1.19.5
    Uninstalling numpy-1.19.5:
      Successfully uninstalled numpy-1.19.5
Successfully installed numpy-1.22.3
    CC stm_gap8_cpx.c
    CC com.c
    CC cpx.c
    ASM  gap8_iet.S
    ASM  startup_gap8.S
    ASM  cluster_core.S
    ASM  asm_util.S
    ASM  port_asm.S
...
    CC  event_groups.c
    CC  stream_buffer.c
    CC  FreeRTOS_util.c
gapy --target=ai_deck --platform=board --work-dir=/module/examples/other/stm_gap8_cpx/BUILD/GAP8_V2/GCC_RISCV_FREERTOS    run --image --binary=/module/examples/other/stm_gap8_cpx/BUILD/GAP8_V2/GCC_RISCV_FREERTOS/stm_gap8_cpx 
gapy --target=ai_deck --platform=board --work-dir=/module/examples/other/stm_gap8_cpx/BUILD/GAP8_V2/GCC_RISCV_FREERTOS    run --flash --binary=/module/examples/other/stm_gap8_cpx/BUILD/GAP8_V2/GCC_RISCV_FREERTOS/stm_gap8_cpx 
gapy --target=ai_deck  gen_flash_image --boot-loader /module/examples/other/stm_gap8_cpx/BUILD/GAP8_V2/GCC_RISCV_FREERTOS/stm_gap8_cpx  -o /module/examples/other/stm_gap8_cpx/BUILD/GAP8_V2/GCC_RISCV_FREERTOS/flash.img 
```

**Reply by LucasTheIntern** [reply]
Got it, I was missing the cd command!

Thank you!

**Reply by knmcguire** [reply]
Great! Let us know if you have any more problems! Since you already selected an answer a new discussion thread is better for that.

**Comment by Hamedamz** [comment]
Hello,

I have an issue when I try to build the stm-gap8 cpx using the following command:

```
~/aideck-gap8-examples$ docker run --rm -v ${PWD}:/module bitcraze/aideck tools/build/make-example examples/other/stm_gap8_cpx all
Collecting numpy==1.22.3
  Using cached numpy-1.22.3-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (16.8 MB)
Installing collected packages: numpy
  Attempting uninstall: numpy
ERROR: imageio 2.17.0 has requirement pillow>=8.3.2, but you'll have pillow 7.2.0 which is incompatible.
    Found existing installation: numpy 1.24.3
    Uninstalling numpy-1.24.3:
      Successfully uninstalled numpy-1.24.3
Successfully installed numpy-1.22.3
gapy --target=ai_deck --platform=board --work-dir=/module/examples/other/stm_gap8_cpx/BUILD/GAP8_V2/GCC_RISCV_FREERTOS    run --image --binary=/module/examples/other/stm_gap8_cpx/BUILD/GAP8_V2/GCC_RISCV_FREERTOS/stm_gap8_cpx  
gapy --target=ai_deck --platform=board --work-dir=/module/examples/other/stm_gap8_cpx/BUILD/GAP8_V2/GCC_RISCV_FREERTOS    run --flash --binary=/module/examples/other/stm_gap8_cpx/BUILD/GAP8_V2/GCC_RISCV_FREERTOS/stm_gap8_cpx  
TARGET create
Open On-Chip Debugger 0.10.0+dev-gb84d97ec4 (2022-04-20-14:26)
Licensed under GNU GPL v2
For bug reports, read
	http://openocd.org/doc/doxygen/bugs.html
debug_level: 0

DEPRECATED! use 'adapter speed' not 'adapter_khz'
DEPRECATED! use 'adapter driver' not 'interface'
GAP8 INIT TARGET
Error: libusb_open() failed with LIBUSB_ERROR_NO_DEVICE
Error: no device found
Error: unable to open ftdi device with vid 15ba, pid 002a, description 'Olimex OpenOCD JTAG ARM-USB-TINY-H', serial '*' at bus location '*'

Flashing image with command:
gap8-openocd -d0 -c "gdb_port disabled; telnet_port disabled; tcl_port disabled" -c "script interface/ftdi/olimex-arm-usb-tiny-h.cfg; script target/gap8revb.tcl; script tcl/flash_image.tcl; script tcl/jtag_boot.tcl; gap_flash_raw_hyper /module/examples/other/stm_gap8_cpx/BUILD/GAP8_V2/GCC_RISCV_FREERTOS/target.board.devices.flash.img 50208 /gap_sdk/utils/openocd_tools; exit;"
Traceback (most recent call last):
  File "/gap_sdk/utils/gapy/gapy", line 162, in <module>
    main()
  File "/gap_sdk/utils/gapy/gapy", line 152, in main
    operationFunc(args, config, system)
  File "/gap_sdk/utils/gapy/run.py", line 189, in operationFunc
    raise RuntimeError('Runner has failed with value: %d' % status)
RuntimeError: Runner has failed with value: -1
make: *** [/gap_sdk/rtos/freeRTOS/vendors/gwt/rules/freeRTOS_rules.mk:442: flash_fs] Error 1
```

Do you know what is the problem?

**Reply by knmcguire** [reply]
Could you start a new discussion thread? This one was about something different and has already been resolved.
